todo_app/tests/test_requests.py

```diff
@@ -1,49 +1,98 @@
+        Item.item_description: "My cool task 1",
+        Item.item_status: BoardStatus._TODO, 
+        Item.item_last_updated: datetime.utcnow()
+    }, 
+    {
+        Item.item_title: "Task 2",
+        Item.item_due_date: "14/03/2021",
+        Item.item_description: "My cool task 2",
+        Item.item_status: BoardStatus._TODO, 
+        Item.item_last_updated: datetime.utcnow()
+    }]
+   
+new_task = {
+        Item.item_title: "Task 3",
+        Item.item_due_date: "17/03/2021",
+        Item.item_description: "My cool task 3"
+    }
 
-def mock_responses(method, url, **kwargs):
-    response = Mock()
-    if url.startswith('https://api.trello.com/1/boards/%s/cards' % mockresponses.board_id):
-        response.json.return_value = mockresponses.cards
-        return response
-        
-    elif url.startswith('https://api.trello.com/1/members/%s/boards' % os.getenv('TRELLO_USER')):
-        response.json.return_value = mockresponses.boards
-        return response
+task_to_be_deleted = {
+        Item.item_title: "Task 4",
+        Item.item_due_date: "22/03/2021",
+        Item.item_description: "My cool task 4",
+        Item.item_status: BoardStatus._TODO
+    }
 
+task_to_be_moved = {
+        Item.item_title: "Task 5",
+        Item.item_due_date: "29/03/2021",
+        Item.item_description: "My cool task 5",
+        Item.item_status: BoardStatus._TODO
+    }
 
-    return None
-
+def get_db_collection():
+    db_conn = os.getenv('DB_CONNECTION_STRING')
+    client = pymongo.MongoClient(db_conn)
+    return client.todoapp['tasks']
 
 @pytest.fixture
-def client():
+def app_client():
     # Use our test integration config instead of the 'real' version
     file_path = find_dotenv('.env.test')
     load_dotenv(file_path, override=True)
 
-    # Create the new app.
-    test_app = app.create_app()
-    # Use the app to create a test_client that can be used in our tests.
-    with test_app.test_client() as client:
-        yield client  
+    # Initialise mongomock here.
+    with mongomock.patch(servers=(('fakemongo', 27017),)):
+        # Create the new app.
+        test_app = app.create_app()
+        # Use the app to create a test_client that can be used in our tests.
+        with test_app.test_client() as client:
+            yield client
 
 
-@patch('requests.request')
-def test_index_page(mock_get_requests, client):
-    mock_get_requests.side_effect = mock_responses
+def test_show_tasks(app_client):
+    tasks = get_db_collection() 
+    tasks.insert_many(initial_mock_tasks)
+    
+    response = app_client.get('/')
+    assert response.status_code == 200 
+    assert tasks.find({Item.item_title: "Task 1"}) is not None
 
-    response = client.get('/')
+def test_add_task(app_client):
+    tasks = get_db_collection() 
+
+    response = app_client.post('/add', data=dict(new_task))
     assert response.status_code == 200 
+    assert tasks.find({Item.item_title: "Task 3"}) is not None
```

Comment:
Using `new_task['Item.item_title]` rather than just `"Task 3"` would make the test a bit clearer (/more robust)

Also there's no real reason for `new_task` to be defined at the top of the file rather than inside this one test where it's used.

Reply:
Done - good point